### PR TITLE
Prevent extra attributes in component configs

### PR DESCRIPTION
**Pull request by schustmi**
## Describe changes
Prevent extra attributes in the config classes of our stack components. Previously, this was defined in the `StackComponentConfig` parent class, but due to the inheritance order lots of our stack components allowed extra attributes. This PR fixes that by making sure extras are not allowed during stack component config validation.
- I also thought about changing the inheritance order of all config classes. However I think it is very unreliable and might lead to newly created stack components having the same issues again. It would also lead to crashes for existing stack components that were registered with a "faulty" config, which my solution does not. It will only fail if users now try to update such a faulty component, at which point they will have to remove the extra attributes.
- Potential issues with my implementation: If some stack component would require `extra="allow"` in their configuration for some reason, this would now be ignored and extras will not be allowed. I can't think of any case where this would be necessary though.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [ ] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/stack/utils.py [hunk 1]
```diff
@@ -75,7 +75,16 @@ def validate_stack_component_config(
             return None
         raise
 
-    configuration = flavor_class.config_class(**configuration_dict)
+    config_class = flavor_class.config_class
+    # Make sure extras are forbidden for the config class. Due to inheritance
```
Comment (bcdurak) [inline src/zenml/stack/utils.py:79]: As you mentioned in the description, for whatever reason, if a user sets the `extra` to `allow`, perhaps we can use a warning or a debug message here.

**Review by bcdurak — APPROVED** [review]
Looks good!

If I understood correctly, people will still be able to use their `Stack` and `StackComponent` objects in Python with "faulty" configs for the time being. This is just blocking them from creating new "faulty" components or updating them in a bad way.  We should maybe look into incorporating this into our design at some point in the future. 

Added some small comments, feel free to ignore.

File: src/zenml/stack/utils.py (hunk 1 above)
Comment (schustmi) [reply src/zenml/stack/utils.py:79]: The problem is, I can't even detect that right? Whether someone did it explicitly or whether it happened due to inheritance of `BaseSettings`

Comment (bcdurak) [reply src/zenml/stack/utils.py:79]: True. Feel free to skip it.

**Pull request merged**